tests/streaming_test.py

```diff
@@ -322,3 +328,103 @@ def _make_target(accum_a: AccumA, accum_b: AccumB) -> Target:
     assert sc.identical(result[Target], sc.scalar(2 * 6.0 / 15.0))
     # Outputs are not cached.
     assert _make_target.call_count == 3
+
+
+def test_StreamProcessor_does_not_reuse_dynamic_keys() -> None:
+    base_workflow = sciline.Pipeline(
+        (make_accum_a, make_accum_b, make_target), params={StaticA: 2.0}
+    )
+    orig_workflow = base_workflow.copy()
+
+    streaming_wf = streaming.StreamProcessor(
+        base_workflow=base_workflow,
+        dynamic_keys=(DynamicA, DynamicB),
+        target_keys=(Target,),
+        accumulators=(AccumA, AccumB),
+    )
+    result = streaming_wf.add_chunk({DynamicA: sc.scalar(1), DynamicB: sc.scalar(4)})
+    assert sc.identical(result[Target], sc.scalar(2 * 1.0 / 4.0))
+    result = streaming_wf.add_chunk({DynamicA: sc.scalar(2)})  # Only A
+    assert not sc.identical(result[Target], sc.scalar(2 * 3.0 / 8.0))
```

Comment:
Not sure why we need to check that the result is not equal to `sc.scalar(2 * 3.0 / 8.0)` here, if we are checking against the correct expected value just below?

Reply:
"documenting" what happened before the bug was fixed, otherwise not much.